### PR TITLE
chore: change to use uds-common test action

### PR DESCRIPTION
**Pull request by ericwyles**
## Description
Updated the test workflow to use new test action from uds-common

Updated other uds-common references to the latest version.

## Related Issue

Relates to https://github.com/defenseunicorns/uds-software-factory/issues/17

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-mattermost/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .github/workflows/tag-and-release.yaml [hunk 1]
```diff
@@ -39,10 +39,10 @@ jobs:
...
-
-      - name: Iron Bank Login
-        run: uds zarf tools registry login -u "${{secrets.IRON_BANK_ROBOT_USERNAME}}" -p "${{secrets.IRON_BANK_ROBOT_PASSWORD}}" registry1.dso.mil
+        uses: defenseunicorns/uds-common/.github/actions/setup@442bae718050ea9a47254851a45632aabeb13b17
+        with:
+          username: ${{secrets.IRON_BANK_ROBOT_USERNAME}}
+          password: ${{secrets.IRON_BANK_ROBOT_PASSWORD}}
     
       - name: Login to GHCR
         uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3
```
Comment (Racer159) [inline .github/workflows/tag-and-release.yaml:48]: this should also have save-logs after line `55` (it is really useful when things fail)

**Review by Racer159 — CHANGES_REQUESTED** [review]
partial review before mtg.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -30,10 +30,10 @@ permissions:
   contents: read
```
Comment (Racer159) [inline .github/workflows/test.yaml:30]: this needs `milestoned` as well (assuming we want to keep that cc @zachariahmiller )

File: tasks.yaml [hunk 3]
```diff
@@ -12,21 +12,27 @@ tasks:
       - task: create-test-bundle
```
Comment (Racer159) [inline tasks.yaml:12]: We might as well standardize these includes to pull from uds-common v0.2.2 like gitlab

**Review by Racer159 — CHANGES_REQUESTED** [review]
rest of comments

File: .github/workflows/tag-and-release.yaml (hunk 1 above)
Comment (ericwyles) [reply .github/workflows/tag-and-release.yaml:48]: Added. Used the same suffix pattern from test.yaml.

File: tasks.yaml (hunk 3 above)
Comment (ericwyles) [reply tasks.yaml:12]: @Racer159 For clarity here... are you suggesting:

1. Update the existing "includes" version numbers to v0.2.2
2. Refactor everything from the 'tasks' folder to also be pulled from uds-common so there are no includes from local paths
3. Something else?

Or all of the above? 

#1 is simple. 

#2 might be nice but I'd probably want to do it in a separate change just to get the benefit of this first round merged while I wrangle what has to happen to move everything out to uds-common for the tasks. As it stands some of these have different tasks that have to execute and I'm not sure how much can be made common on that side just yet. I kept that main 4 task names in the base tasks.yaml as the entry point for CI but they diverge some within the implementation of those 4.

File: .github/workflows/test.yaml (hunk 2 above)
Comment (ericwyles) [reply .github/workflows/test.yaml:32]: Added this.

File: tasks.yaml (hunk 3 above)
Comment (Racer159) [reply tasks.yaml:12]: number 1, sorry

Comment (Racer159) [reply tasks.yaml:12]: This will get caught with renovate later but just wanted to mention it as a current diff

Comment (ericwyles) [reply tasks.yaml:12]: I went ahead and updated it manually. 👍

**Comment by ericwyles** [comment]
@Racer159 I think this one is ready for another look. It'll need updates from an owner for the 'Required' checks that are listed as 'Expected' now because names changed with the consolidation.

**Review by zachariahmiller — DISMISSED** [review]

**Comment by zachariahmiller** [comment]
@ericwyles updated the required checks for the name change not sure why this one is still showing the old ones... 

anyways this had the docs shim merged and as such will also need the job name for that updated to match the test. https://github.com/defenseunicorns/uds-package-mattermost/blob/5d603fd284f8c366daffff2326b7813669f7c692/.github/workflows/ci-docs-shim.yaml#L32
hopefully fixing that will reset the required checks to the current ones 🤔

**Comment by zachariahmiller** [comment]
I actually just went and did it so we dont have to think about it tomorrow not sure about the checks though. Will send over to @Racer159 for approval since i now have last commit 😢

**Review by Racer159 — APPROVED** [review]
lgtm (the save logs doesn't have a trailing new line but that's probably fine for now and really a nit)

**Pull request merged**